src/install/install.zig

```diff
@@ -3253,7 +3409,7 @@ pub const PackageManager = struct {
             }
 
             // allow overriding all dependencies unless the dependency is coming directly from an alias, "npm:<this dep>"
-            if (dependency.version.tag != .npm or !dependency.version.value.npm.is_alias) {
+            if (dependency.version.tag != .npm or !dependency.version.value.npm.is_alias and this.lockfile.overrides.map.count() > 0) {
```

Comment:
maybe we could add an inline fn for `this.lockfile.hasOverrides()` that way if zig changes .count() in the future we don't have to change as many call sites?